cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,475 @@
+    """In these tests, chargeback reports are filtered on VM owner.So,VMs have to be
+    assigned ownership.
+    """
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    if not provider.mgmt.does_vm_exist(vm_name):
+        pytest.skip("Skipping test, cu-24x7 VM does not exist")
+    if not provider.mgmt.is_vm_running(vm_name):
+        provider.mgmt.start_vm(vm_name)
+        provider.mgmt.wait_vm_running(vm_name)
+
+    group_collection = appliance.collections.groups
+    cb_group = group_collection.instantiate(description='EvmGroup-user')
+
+    vm = VM.factory(vm_name, provider)
+    user = appliance.collections.users.create(
+        name='{}_{}'.format(provider.name, fauxfactory.gen_alphanumeric()),
+        credential=Credential(principal='uid' + '{}'.format(fauxfactory.gen_alphanumeric()),
+            secret='secret'),
+        email='abc@example.com',
+        groups=cb_group,
+        cost_center='Workload',
+        value_assign='Database')
+    vm.set_ownership(user=user.name)
+    logger.info('Assigned VM OWNERSHIP for {} running on {}'.format(vm_name, provider.name))
+
+    yield user.name
+
+    vm.unset_ownership()
+    if user:
+        user.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def enable_candu(appliance):
+    """C&U data collection consumes a lot of memory and CPU.So, we are disabling some server roles
+    that are not needed for Chargeback reporting.
+    """
+    candu = appliance.collections.candus
+    server_info = appliance.server.settings
+    original_roles = server_info.server_roles_db
+    server_info.enable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector', 'ems_metrics_processor')
+    server_info.disable_server_roles('automate', 'smartstate')
+    candu.enable_all()
+
+    yield
+
+    server_info.update_server_roles_db(original_roles)
+    candu.disable_all()
+
+
+@pytest.yield_fixture(scope="module")
+def assign_custom_rate(new_compute_rate):
+    """Assign custom Compute rate to the Enterprise and then queue the Chargeback report."""
+    description = new_compute_rate
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': description}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+    logger.info('Assigning CUSTOM Compute rate')
+
+    yield
+
+    # Resetting the Chargeback rate assignment
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': '<Nothing>'}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+
+
+def verify_records_rollups_table(appliance, provider):
+    """Verify that hourly rollups are present in the metric_rollups table."""
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    rollups = appliance.db.client['metric_rollups']
+
+    with appliance.db.client.transaction:
```

Comment:
REQUIRED: please don't use db where it is possible. we should get rid of db or ssh calls wherever it is possible. please use rest api instead.

app.rest_api.collections.metric_rollups
Out[4]: <Collection u'metric_rollups' (u'Metric Rollups')>

Reply:
I'm doing inner join queries on two different tables in two places and an inner join query on three different tables in another place, hence using db.